pkg/controller/operands/kubevirt.go

```diff
@@ -450,29 +450,36 @@ func NewKubeVirtWithNameOnly(hc *hcov1beta1.HyperConverged, opts ...string) *kub
 	}
 }
 
-func hcoConfig2KvConfig(hcoConfig hcov1beta1.HyperConvergedConfig) *kubevirtcorev1.ComponentConfig {
-	if hcoConfig.NodePlacement != nil {
+func hcoConfig2KvConfig(hcoConfig hcov1beta1.HyperConvergedConfig, infrastructureHighlyAvailable bool) *kubevirtcorev1.ComponentConfig {
+	if hcoConfig.NodePlacement != nil || !infrastructureHighlyAvailable {
```

Comment:
not sure it's so important, but I think it's the two cases in this condition handled desperately below with almost no common code (beside the new  `kvConfig` var and the return statement). Maybe we ca save one layer of nesting and make the sonarCloud happier.